modules/subnets/variables.tf

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "The ID of the project where subnets will be created"
+}
+
+variable "network_name" {
+  description = "The name of the network where subnets will be created"
+}
+
+variable "subnets" {
+  type        = list(map(string))
```

Comment:
This should be updated to a list of objects so the expected structure can be documented and enforced.

Reply:
@aaron-lane This has been extensively discussed, but the conclusion we reached is that until objects support defaults we will stick to maps.